pkg/storage/tsdb/block/fetcher.go

```diff
@@ -534,6 +549,25 @@ func (f *MetaFetcher) countCached() int {
 	return len(f.cached)
 }
 
+// FetchRequestedBlocks fetches metadata for specific block IDs without iterating through the bucket
+// Assumes metas exist and returns an error if any blocks fail to download.
+func (f *MetaFetcher) FetchRequestedBlocks(ctx context.Context, blockIDs []ulid.ULID) (map[ulid.ULID]*Meta, error) {
+	if len(blockIDs) == 0 {
+		return nil, errNoBlocksProvided
+	}
+
+	metas, partial, err := f.fetch(ctx, false, blockIDs)
```

Comment:
The `false` here is saying we don't want to skip blocks marked for deletion. I remember discussing with you whether we should be listing markers or not or trusting the planned job. Maybe we could probe one block or something as a compromise between the two.